code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm

```diff
@@ -21,12 +21,28 @@
 		to_chat(caller, "<span class='warning'>Our hive currently has no burrowed to call forth!</span>")
 		return FALSE
 	RegisterSignal(shrike_hive, list(COMSIG_HIVE_XENO_MOTHER_PRE_CHECK, COMSIG_HIVE_XENO_MOTHER_CHECK), .proc/is_burrowed_larva_host)
+	RegisterSignal(shrike_hive, list(COMSIG_HIVE_XENO_MOTHER_CALL_BURROWED, COMSIG_HIVE_XENO_MOTHER_CHECK), .proc/is_burrowed_larva_host)
 
 	playsound(caller,'sound/magic/invoke_general.ogg', 75, 1)
 	new /obj/effect/temp_visual/telekinesis(get_turf(caller))
 	caller.visible_message("<span class='xenowarning'>A strange buzzing hum starts to emanate from \the [caller]!</span>", \
 	"<span class='xenodanger'>We call forth the larvas to rise from their slumber!</span>")
-	notify_ghosts("\The <b>[caller]</b> is calling for the burrowed larvas to wake up!", enter_link = "join_larva=1", enter_text = "Join as Larva", source = caller, action = NOTIFY_JOIN_AS_LARVA)
+
+	var/burrowed = shrike_hive.stored_larva
+	var/total = 0
+	for(var/i in 1 to burrowed)
+		var/mob/M = get_alien_candidate()
+		if(!M)
+			break
+
+		var/list/possible_mothers = list()
+		SEND_SIGNAL(src, COMSIG_HIVE_XENO_MOTHER_CALL_BURROWED, possible_mothers) //List variable passed by reference, and hopefully populated.

```

Comment:
This can be done a single time, before the `for` loop.
There's no `sleep` or `stoplag` in between after all.

Reply:
Actually just use the Shrike as a mother. No need to pick others.